### ISSUE TITLE
Error parsing class method

### ISSUE THREAD

**Issue by jbeezley** [tunnckoCore/parse-function#30]
I've run into a strange edge case on the most recent version trying to parse a class method that contains an arrow function.  Here is a minimal example:
```javascript
const parseFunction = require('parse-function');
const app = parseFunction({ecmaVersion: 2017});

class A {
  func () {
    return () => 0;
  }
}

const a = new A();
var p = app.parse(a.func);
console.log(p.args);
```
This results in
```javascript
./node_modules/parse-function/node_modules/babylon/lib/index.js:4335
  throw err;
  ^

SyntaxError: Unexpected token (1:7)
    at Parser.pp$5.raise (./node_modules/parse-function/node_modules/babylon/lib/index.js:4332:13)
    at Parser.pp.unexpected (./node_modules/parse-function/node_modules/babylon/lib/index.js:1689:8)
    at Parser.pp$3.getExpression (./node_modules/parse-function/node_modules/babylon/lib/index.js:3178:10)
    at Object.parseExpression (./node_modules/parse-function/node_modules/babylon/lib/index.js:6982:17)
    at Object.getNode (./node_modules/parse-function/lib/utils.js:90:18)
    at Object.parse (./node_modules/parse-function/index.js:126:24)
    at Object.<anonymous> (./class_method.js:12:13)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
```
Removing the arrow function, it works as expected.

I'm not sure if this is a babylon bug or not because the string representation of the function is
```javascript
func() {
    return () => 0;
  }
```
which I don't think is a valid expression on its own.  This can be worked around by catching the error and trying again with `'function ' + fn.toString()`, but I'm not sure if that's the best approach.  I can create an issue upstream if you think it would be more appropriate to fix there.

**Comment by tunnckoCore** [comment]
Strange. Not make sense. Maybe it seems invalid to babylon. Can you try with the other parsers?

There is actually a test about object methods, so it not make sense to not work for class methods, since they are the same thing.

I'll review what happens, thanks for the report.

**Comment by tunnckoCore** [comment]
Actually, yea. [index.js#L120](https://github.com/tunnckoCore/parse-function/blob/master/index.js#L120) is the problem. damn.

edit: One more proof the 100% coverage is never enough.

**Comment by tunnckoCore** [comment]
Published as v4.0.1 and will publish 4.0.2 for #31

**Comment by jbeezley** [comment]
It's working great now.  Thanks!
